Zend/zend_ini.c

```diff
@@ -536,6 +538,75 @@ ZEND_API bool zend_ini_parse_bool(zend_string *str)
 	}
 }
 
+ZEND_API zend_long zend_ini_parse_quantity(zend_string *value, zend_string **errstr) /* {{{ */
+{
+	char *digits_end = NULL;
+	char *str = ZSTR_VAL(value);
+	size_t str_len = ZSTR_LEN(value);
+
+	/* Ignore trailing whitespace */
+	while (str_len && ZEND_IS_WHITESPACE(str[str_len-1])) --str_len;
+	if (!str_len) {
+		*errstr = NULL;
+		return 0;
+	}
+
+	/* Perform following multiplications on unsigned to avoid overflow UB.
+	 * For now overflow is silently ignored -- not clear what else can be
+	 * done here, especially as the final result of this function may be
+	 * used in an unsigned context (e.g. "memory_limit=3G", which overflows
+	 * zend_long on 32-bit, but not size_t). */
+	zend_ulong retval = (zend_ulong) ZEND_STRTOL(str, &digits_end, 0);
+
+	if (digits_end == str) {
+		*errstr = zend_strpprintf(0, "Invalid numeric string '%.*s': no valid leading digits, interpreting as '0' for backwards compatibility",
```

Comment:
Can we not use "numeric string" in this context? As from what I see ``"1.0"`` is invalid, which *is* a numeric string. Moreover a numeric string in standard PHP cannot have a quantity modifier behind.

Maybe use ``Invalid INI quantity``?